skyfield/positionlib.py

```diff
@@ -152,18 +151,61 @@ def separation_from(self, another_icrf):
         c = dots(u1, u2)
         return Angle(radians=arccos(clip(c, -1.0, 1.0)))
 
-    def ecliptic_position(self):
-        """Compute J2000 ecliptic coordinates (x, y, z)"""
-        vector = _ECLIPJ2000.dot(self.position.au)
-        return Distance(vector)
+    def ecliptic_xyz(self, epoch=None):
+        """Compute J2000 ecliptic coordinates (x, y, z)
 
-    def ecliptic_latlon(self):
-        """Compute J2000 ecliptic coordinates (lat, lon, distance)"""
-        vector = _ECLIPJ2000.dot(self.position.au)
-        d, lat, lon = to_polar(vector)
-        return (Angle(radians=lat, signed=True),
-                Angle(radians=lon),
-                Distance(au=d))
+        If you instead want the coordinates referenced to the dynamical
+        system defined by the Earth's true equator and equinox, provide
+        an epoch time.
+        """
+        lat, lon, d = self.ecliptic_latlon(epoch)
+        vector = from_polar(d.au, lat.radians, lon.radians)
```

Comment:
It feels a little bit roundabout to have `ecliptic_latlon()` (a) compute rectangular coordinates, (b) convert them to polar, and then (c) in this routine convert them back to rectangular. Could you move the `einsum` code that computes the rectangular coordinates up into this routine instead, and then have the polar routine call this one?

Also, can you eliminate the `compute_ECLIP()` function and simply call `earth_tilt()` and `rot_x()` right here?

Reply:
I think I will be able to do it.

Reply:
I eliminated compute_ECLIP, however when i reorder stuff so that ecliptic_latlon calls ecliptic_xyz and converts vector to polar, I end up with this:
File "main.py", line 52, in <module>
    lat, lon, d = apparent.ecliptic_latlon(epoch='date')
File "python-skyfield/skyfield/positionlib.py", line 256, in ecliptic_latlon
    d, lat, lon = to_polar(vector)
File "python-skyfield/skyfield/functions.py", line 36, in to_polar
    r = length_of(xyz)
File "python-skyfield/skyfield/functions.py", line 24, in length_of
    return sqrt((xyz * xyz).sum(axis=0))
TypeError: unsupported operand type(s) for *: 'Distance' and 'Distance'

Reply:
My guess is that you don't want the Distance object, but the `.au` attribute inside of it, which is the actual xyz position as a NumPy array. So maybe `to_polar(vector.au)`?